io/net/bootlist_test.py

```diff
@@ -0,0 +1,160 @@
+        '''
+        cmd = "bootlist -m normal -r > /tmp/normal"
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("unable to write normal bootlist in to the file")
+
+    def set_original_normal_bootlist(self):
+        '''
+        set the original bootlist
+        '''
+        cmd = "bootlist -m normal -f /tmp/normal"
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("failed to set default normal bootlist")
+
+    def service_bootlist_file(self):
+        '''
+        default sevice bootlist write in to a file
+        '''
+        cmd = "bootlist -m service -r > /tmp/service"
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("unable to write normal bootlist in to the file")
+
+    def set_original_service_bootlist(self):
+        '''
+        set the original service bootlist
+        '''
+        cmd = "bootlist -m service -f /tmp/service"
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("failed to set default service bootlist")
+
+    @staticmethod
+    def display_normal_logical_device():
+        '''
+        Displaying normal bootlist entries as logical device name
+        '''
+        cmd = "bootlist -m normal -o"
+        for line in process.system_output(cmd, shell=True,
+                                          ignore_status=True).splitlines():
+            return line.split()[-1]
+
+    @staticmethod
+    def display_normal_firmware_device():
+        '''
+        Display normal bootlist entries as open firmware devices
+        '''
+        cmd = "bootlist -m normal -r"
+        for line in process.system_output(cmd, shell=True,
+                                          ignore_status=True).splitlines():
+            return line.split()[-1]
+
+    @staticmethod
+    def display_service_logical_device():
+        '''
+        Displaying service bootlist entries as logical device name
+        '''
+        cmd = "bootlist -m service -o"
+        for line in process.system_output(cmd, shell=True,
+                                          ignore_status=True).splitlines():
+            return line.split()[-1]
+
+    @staticmethod
+    def display_service_firmware_device():
+        '''
+        Display service bootlist entries as open firmware devices
+        '''
+        cmd = "bootlist -m service -r"
+        for line in process.system_output(cmd, shell=True,
+                                          ignore_status=True).splitlines():
+            return line.split()[-1]
+
+    def test_normal_mode(self):
+        '''
+        To make a boot list for Normal mode
+        '''
+        self.normal_bootlist_file()
+        cmd = "bootlist -m normal %s" % self.interfaces
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            self.fail("is not supported on the Powernv Host")
```

Comment:
1. PowerNV check can be done at the start.
2. This failure need not be PowerNV. It could be anything else also, right ?